fix(nvidia): switch back to ngl renderer

alexvojproc opened the pull request:
adds a /usr/lib/environment.d drop-in for nvidia images to set the gtk renderer to `ngl`, rather than the default `vulkan`

- bazzite have patched but don't seem to be tracking it: https://github.com/ublue-os/bazzite/commit/bb0e8f0d9cfe26efa56eed0fa75b7d1e8c110c46
- haven't found an upstream issue yet 🫤 
  - nvidia claim to have fixed a similar issue in [580.65.06](https://www.nvidia.com/en-us/drivers/details/251355/), but apparently still a problem on secureblue 580.76.05
- builds and behaves as expected, has fixed issue for two users

---

This was the previous default, so it should be safe. This should be reverted when the upstream issue is resolved.

secureblue issue: <https://github.com/secureblue/secureblue/issues/1293>

HastD commented on files/scripts/setgtknvidiaenvironment.sh:
```diff
@@ -0,0 +1,17 @@
+
+set -oue pipefail
+
+echo 'GSK_RENDERER=ngl' > /usr/lib/environment.d/40-nvidia-gtk-renderer.conf
```
Minor nitpick, but according to [environment.d(5)](https://www.man7.org/linux/man-pages/man5/environment.d.5.html):

> It is recommended to prefix all filenames with a two-digit number and a dash to simplify the ordering. It is recommended to use the range 10-40 for configuration files in /usr/ [...]

We already have a `40-hardened_malloc.conf` drop-in, so I'd suggest using a different prefix in the recommended range.

alexvojproc replied:
> We already have a `40-hardened_malloc.conf` drop-in, so I'd suggest using a different prefix in the recommended range.

@HastD Ah ok, TIL! I knew about the 10-40 range, but I thought it was fine to have multiple files with the same prefix, assuming you'd never want to insert anything between them. I guess that's needless, I'll move it 😄

alexvojproc committed:
fix(nvidia): switch back to ngl renderer

This was the previous default, so it should be safe. This should be reverted when the upstream issue is resolved.

secureblue issue: <https://github.com/secureblue/secureblue/issues/1293>

alexvojproc commented:
Seems this could be fixed in the next release: https://forums.developer.nvidia.com/t/580-release-feedback-discussion/341205/108

RoyalOughtness committed:
Merge branch 'live' into nvidianglfix
